### DISCUSSION TITLE
Spatialite compatibility

### DISCUSSION THREAD

**Discussion by triglav** [prisma/prisma#19713] [Q&A]
Is there a way to use Prisma with [spatialite](https://www.gaia-gis.it/fossil/libspatialite/index)?

The database needs to be typically initialized with the following:
```sql
SELECT load_extension("mod_spatialite");
SELECT InitSpatialMetaData();
```
This creates a lot of tables and views, etc.
Is there a way to ignore these in the schema or such?

**Comment by nurul3101** [comment]
Hello @triglav 👋 

We don't have support for Spatial types natively yet:

- https://github.com/prisma/prisma/issues/1798

Once you have installed `mod_spatialite` extension to the database, you can create a migration file with `--create-only` flag and then edit the migration file to include these two SELECT statements. Here's a [reference](https://www.prisma.io/docs/guides/migrate/developing-with-prisma-migrate/customizing-migrations#how-to-edit-a-migration-file) on how to customize a migration.

You can use [@@ignore](https://www.prisma.io/docs/reference/api-reference/prisma-schema-reference#ignore-1) attribute to ignore certain models in the schema. This won't prevent these tables from appearing in the schema file but they won't be surfaced in the generated Prisma Client.

If this answers your question, it would be great if you could mark this Discussion as answered to indicate that it has been resolved.

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Comment by ludralph** [comment]
Hi @triglav 👋 

You will need to have `Spatialite` installed with the SQLite database in order to use it. Once this is installed, you can customize a migration in order to use it. To customize a migration to include the `Spatialite` feature:

1. Use the --create-only flag to generate a new migration without applying it:

```
npx prisma migrate dev --create-only
```

2. Open the generated migration.sql file and add the code below
```sql
SELECT load_extension("mod_spatialite");
SELECT InitSpatialMetaData();
```
3. Apply the migration using the command below
```
npx prisma migrate dev
```
This should enable Spatialite to be used with SQlite within Prisma.


> This creates a lot of tables and views, etc.
Is there a way to ignore these in the schema or such?

Just to clarify, do you want to ignore the tables when you introspect them? If that's what you want to achieve, we do have a feature request for it here:

- https://github.com/prisma/prisma/issues/807

You can add a 👍  and write down your use case in that issue to help our engineers prioritize it. 

If this answers your question, it would be great if you could **mark this Discussion as answered** to indicate that it has been resolved. 

Otherwise please let us know how else we can help you further or close the Discussion if it was resolved in some other way 🙏

**Reply by triglav** [reply]
Thank you, I have actually already tried creating a custom migration for this.
It results in an error however:
```
Error: P3018

A migration failed to apply. New migrations cannot be applied before the error is recovered from. Read more about how to resolve migration issues in a production database: https://pris.ly/d/migrate-resolve

Migration name: 20230612162500_init_spatialite

Database error code: 1

Database error:
not authorized
```

**Reply by andreaspapav** [reply]
I am stuck at the exact same error. when using the sqlite3 console i can load the spatialite just fine but when including it in the migration i get the not authorized error

**Reply by ludralph** [reply]
Hi @andreaspapav 

Since we don't have support for Spatial types natively yet, it's possible to see these type of errors.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;
